### PR TITLE
Update bicep parameter file #159

### PR DESCRIPTION
**Pull request by nilsson-jens**
Created possible solution for updating Parameter files according to issue #159.
Added cmdlet Update-BicepParameterFile and updated Docks / Help Files.

### CODE REVIEW

**Review by SimonWahlin — COMMENTED** [review]
Thank you so much for your contribution

I have left a few suggestions for improvements.

**Comment by StefanIvemo** [comment]
I've been testing this and it works great! But I'm starting to think that we need to add the -Parameters switch that we have for `New-BicepParameterFile`. For a new parameter file we default to `-Parameters Required` and I believe we should do the same for update.

Example:

I have a template like this, called `template.bicep`:
```bicep
param name string
param location string = resourceGroup().location
param sku string
param deploy bool = true
param array array = []
```

I generate a parameter using `New-BicepParameterFIle`:
```powershell
New-BicepParameterFile -Path .\template.bicep
```

This will give me a `template.parameters.json` that looks like this:
```json
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "name": {
      "value": ""
    },
    "sku": {
      "value": ""
    }
  }
}
```

If I edit the `template.bicep` file and add one more parameter like this:
```bicep
param name string
param location string = resourceGroup().location
param sku string
param deploy bool = true
param array array = []
param object object
```

And then update the parameter file using `Update-BicepParameterFile`:
```powershell
Update-BicepParameterFile -Path .\template.bicep
```

I get this result in my `template.parameters.json`:
```json
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "name": {
      "value": ""
    },
    "location": {
      "value": ""
    },
    "sku": {
      "value": ""
    },
    "deploy": {
      "value": true
    },
    "array": {
      "value": []
    },
    "object": {
      "value": {}
    }
  }
}
```

What I want to see:
- A `-Parameters` parameter that accepts `Required and All` just like for `New-BicepParameterFile`.
- We should default to `All` just like we do for `New-BicepParameterFile`.


Thoughts @nilsson-jens and @SimonWahlin?

File: Source/Public/Update-BicepParameterFile.ps1 [hunk 1]
```diff
@@ -0,0 +1,75 @@
+function Update-BicepParameterFile {
+    [CmdletBinding()]
+    param (
+        [ValidateNotNullOrEmpty()]
+        [Parameter(Mandatory, Position = 1)]
+        [string]$Path,
```
Comment (StefanIvemo) [inline Source/Public/Update-BicepParameterFile.ps1:6]: Can we add a validate script to `-Path` to only accept `.json` files as input. Accidentally pointed to the `.bicep` file while testing and it broke my template.

File: Source/Public/Update-BicepParameterFile.ps1 [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+        [ValidateNotNullOrEmpty()]
+        [Parameter(Position = 2)]
+        [string]$BicepFile
```
Comment (StefanIvemo) [inline Source/Public/Update-BicepParameterFile.ps1:10]: Same thing here, we should validate that the input to `-BicepFile` is a `.bicep` file.

**Comment by SimonWahlin** [comment]
A -Parameters parameter sounds like a super good idea!

**Comment by StefanIvemo** [comment]
I will move the part about adding a `-Parameters` parameter to a separate issue to be able to merge this.

**Review by StefanIvemo — APPROVED** [review]

**Pull request merged**